libcxx/include/__format/buffer.h

```diff
@@ -722,6 +724,95 @@ class _LIBCPP_TEMPLATE_VIS __allocating_buffer : public __output_buffer<_CharT>
   }
 };
 
+// A buffer that directly writes to the underlying buffer.
+template <class _OutIt, __fmt_char_type _CharT>
```

Comment:
```suggestion
template <contiguous_iterator _OutIt, __fmt_char_type _CharT>
```

That way, you don't need to `__unwrap_iter` below and we can keep the bounds information when there is some.